src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -169,28 +169,24 @@ bool TryGetWorkloadSet(string workloadSetVersion, out WorkloadSet? workloadSet)
                 }
             }
 
+            _installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkVersionBand, _sdkRootPath), "default.json");
+            var installState = InstallStateContents.FromPath(_installStateFilePath);
             if (_workloadSet is null)
             {
-                var installStateFilePath = Path.Combine(WorkloadInstallType.GetInstallStateFolder(_sdkVersionBand, _sdkRootPath), "default.json");
-                if (File.Exists(installStateFilePath))
+                if (!string.IsNullOrEmpty(installState.WorkloadVersion))
                 {
-                    var installState = InstallStateContents.FromPath(installStateFilePath);
-                    if (!string.IsNullOrEmpty(installState.WorkloadVersion))
+                    if (!availableWorkloadSets.TryGetValue(installState.WorkloadVersion!, out _workloadSet))
```

Comment:
Was this change intentional?  Maybe it was a rebase/merge issue.  I think we want to call `TryGetWorkloadSet` here.

```suggestion
                    if (!TryGetWorkloadSet.TryGetValue(installState.WorkloadVersion!, out _workloadSet))
```

Reply:
Ah, yes. Sorry; missed that when I was merging the branches. Will revert.